### PR TITLE
`Programming exercises`: Fix internal server error when multiple tasks have no test cases

### LINKED ISSUE #5475: Internal Server Error when retrieving hints for a programming exercise with a problem statement with wrong test case names
### Describe the bug

I receive an internal server error when I try to update the page after pushing to the exercise repo (FPV homework exercise).
Exercise link: https://artemis.ase.in.tum.de/courses/189/exercises/7426
This page throws a status 500: https://artemis.ase.in.tum.de/api/programming-exercises/7426/exercise-hints/available
See also: https://zulip.in.tum.de/#narrow/stream/1157-FPV-Sheet13/topic/W13.20H02.20-.20Tests.20Server.20failure

### To Reproduce

1. Go to Course 189
2. Push some changes
3. Update the Artemis exercise page
4. Internal server error

### Expected behavior

The program should build and show hints for each one of the subtasks.

### Screenshots

image
image
image

### What browsers are you seeing the problem on?

Other (specify in "Additional context")

### Additional context

Used browser: Brave Browser

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by Hialus**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I added `@PreAuthorize` and checked the course groups for all new REST Calls (security).
- [x] I implemented the changes with a good performance and prevented too many database calls.
- [x] I documented the Java code using JavaDoc style.
### Motivation and Context

Fixes #5475

### Description

This error was caused by multiple tasks having no test cases, leading to an issue when sorting the tasks.
This was fixed by also comparing tasks by name and not only by tests when sorting them.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- 1 Programming Exercise

1. Log in to Artemis
2. Add multiple tasks without (or with wrong) test cases to the problem statement
3. Participate in the exercise as a student and do at least one submission
4. Wait for the submission build
5. No internal server error should occur

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage
Unchanged

### Screenshots

Multiple tasks without test cases:
grafik

### CODE REVIEW

**Review by bassner — APPROVED** [review]
Tested on TS1 - did not get an error on tasks with wrong test cases (looks like in your screenshot).

Code looks good. 👍

**Review by babyygemperor — DISMISSED** [review]
Tested on ts1, works as described

File: src/main/java/de/tum/in/www1/artemis/service/hestia/ProgrammingExerciseTaskService.java [hunk 1]
```diff
@@ -112,7 +112,9 @@ public List<ProgrammingExerciseTask> getSortedTasks(ProgrammingExercise exercise
         var unsortedTasks = programmingExerciseTaskRepository.findByExerciseIdWithTestCases(exercise.getId());
         var sortedExtractedTasks = extractTasks(exercise);
         return sortedExtractedTasks.stream()
-                .map(extractedTask -> unsortedTasks.stream().filter(task -> task.getTestCases().equals(extractedTask.getTestCases())).findFirst().orElse(null))
+                .map(extractedTask -> unsortedTasks.stream()
+                        .filter(task -> task.getTaskName().equals(extractedTask.getTaskName()) && task.getTestCases().equals(extractedTask.getTestCases())).findFirst()
```
Comment (JohannesStoehr) [inline src/main/java/de/tum/in/www1/artemis/service/hestia/ProgrammingExerciseTaskService.java:116]: This is the exact same logic as the equals method in ProgrammingExerciseTask.java

**Review by JohannesStoehr — CHANGES_REQUESTED** [review]

**Review by nairaarnez — DISMISSED** [review]
Tested in TS1, no internal er server error occurred

File: src/main/java/de/tum/in/www1/artemis/service/hestia/ProgrammingExerciseTaskService.java (hunk 1 above)
Comment (Hialus) [reply src/main/java/de/tum/in/www1/artemis/service/hestia/ProgrammingExerciseTaskService.java:116]: It is not. The equals method also compares the id by calling super.equals.

**Review by JohannesStoehr — APPROVED** [review]
Code looks good 👍 
Sorry for the wrong suggestion before

**Review by manuelmanso — APPROVED** [review]
LGTM

**Review by bassner — APPROVED** [review]
Reapprove (nothing changed)

**Pull request merged**